bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/connector/EpsonProjectorSerialConnector.java

```diff
@@ -0,0 +1,220 @@
+                in.close();
+            } catch (IOException e) {
+                logger.debug("Error occurred when closing serial in stream", e);
+            }
+            this.in = null;
+        }
+        if (serialPort != null) {
+            logger.debug("Close serial port");
+            serialPort.close();
+            serialPort.removeEventListener();
+            this.serialPort = null;
+        }
+
+        logger.debug("Closed");
+    }
+
+    @Override
+    public String sendMessage(String data, int timeout) throws EpsonProjectorException {
+        InputStream in = this.in;
+        OutputStream out = this.out;
+
+        if (in == null || out == null) {
+            connect();
+            in = this.in;
+            out = this.out;
+        }
+
+        try {
+            if (in != null && out != null) {
+                // flush input stream
+                if (in.markSupported()) {
+                    in.reset();
+                } else {
+                    while (in.available() > 0) {
+                        int availableBytes = in.available();
+
+                        if (availableBytes > 0) {
+                            byte[] tmpData = new byte[availableBytes];
+                            in.read(tmpData, 0, availableBytes);
+                        }
+                    }
+                }
+
+                return sendMmsg(data, timeout);
+            } else {
+                return "";
+            }
+        } catch (IOException e) {
+            logger.debug("IO error occurred...reconnect and resend once");
+            disconnect();
+            connect();
+
+            try {
+                return sendMmsg(data, timeout);
+            } catch (IOException e1) {
+                throw new EpsonProjectorException(e);
+            }
+        } catch (Exception e) {
+            throw new EpsonProjectorException(e);
+        }
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent arg0) {
+        try {
+            logger.trace("RXTX library CPU load workaround, sleep forever");
+            Thread.sleep(Long.MAX_VALUE);
+        } catch (InterruptedException e) {
+        }
+    }
+
+    private String sendMmsg(String data, int timeout) throws IOException, EpsonProjectorException {
+        String resp = "";
+
+        InputStream in = this.in;
+        OutputStream out = this.out;
+
+        if (in != null && out != null) {
+            out.write(data.getBytes());
+            out.write("\r\n".getBytes());
+            out.flush();
+
+            long startTime = System.currentTimeMillis();
+            long elapsedTime = 0;
+
+            while (elapsedTime < timeout) {
+                int availableBytes = in.available();
+                if (availableBytes > 0) {
+                    byte[] tmpData = new byte[availableBytes];
+                    int readBytes = in.read(tmpData, 0, availableBytes);
+                    resp = resp.concat(new String(tmpData, 0, readBytes));
```

Comment:
Can you specify the encoding you expect? E.g. StandardCharsets.UTF_8

Reply:
done